File: advanced-programming-methods/labs/toy-interpreter-java/src/model/statement/IfStatement.java
package model.statement;

import model.expression.Expression;

/**
 * The type If statement.
 */
public class IfStatement extends Statement
{
    private Expression condition;

    private Statement consequent;

    private Statement alternative;

    /**
     * Instantiates a new If statement.
     *
     * @param condition   the condition
     * @param consequent  the consequent
     * @param alternative the alternative
     */
    public IfStatement(Expression condition, Statement consequent, Statement alternative)
    {
        this.condition = condition;
        this.consequent = consequent;
        this.alternative = alternative;
    }

    /**
     * Gets condition.
     *
     * @return the condition
     */
    public Expression getCondition()
    {
        return condition;
    }

    /**
     * Gets consequent.
     *
     * @return the consequent
     */
    public Statement getConsequent()
    {
        return consequent;
    }

    /**
     * Gets alternative.
     *
     * @return the alternative
     */
    public Statement getAlternative()
    {
        return alternative;
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public String toString()
    {
        return "IfStatement{" +
            "condition=" + condition +
            ", consequent=" + consequent +
            ", alternative=" + alternative +
            '}';
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public Statement cloneDeep()
    {
        return new IfStatement(condition.cloneDeep(), consequent.cloneDeep(), alternative.cloneDeep());
    }
}

File: advanced-programming-methods/labs/toy-interpreter-sharp/model/statement/IfStatement.cs
using System;

namespace Model
{
    /// <summary>
    /// The type If statement.
    /// </summary>
    [Serializable()]
    public class IfStatement : Statement
    {
        private readonly Expression condition;

        private readonly Statement consequent;

        private readonly Statement alternative;

        /// <summary>
        /// Instantiates a new If statement.
        /// </summary>
        /// <param name="condition"> the condition </param>
        /// <param name="consequent"> the consequent </param>
        /// <param name="alternative"> the alternative </param>
        public IfStatement(Expression condition, Statement consequent, Statement alternative)
        {
            this.condition = condition;
            this.consequent = consequent;
            this.alternative = alternative;
        }

        /// <summary>
        /// Gets condition.
        /// </summary>
        /// <returns> the condition </returns>
        public Expression Condition
        {
            get
            {
                return condition;
            }
        }

        /// <summary>
        /// Gets consequent.
        /// </summary>
        /// <returns> the consequent </returns>
        public Statement Consequent
        {
            get
            {
                return consequent;
            }
        }

        /// <summary>
        /// Gets alternative.
        /// </summary>
        /// <returns> the alternative </returns>
        public Statement Alternative
        {
            get
            {
                return alternative;
            }
        }

        /// <inheritdoc />
        public override string ToString()
        {
            return "IfStatement{" + "condition=" + condition + ", consequent=" + consequent + ", alternative=" + alternative + '}';
        }

        /// <inheritdoc />
        public override Statement CloneDeep()
        {
            return new IfStatement(condition.CloneDeep(), consequent.CloneDeep(), alternative.CloneDeep());
        }
    }
}
